profiling/paper_timing_tables/timing_comparison.py

```diff
@@ -0,0 +1,338 @@
+            )
+
+        ptimes_nopickle = []
+        ptimes_pickle = []
+        ttimes = []
+        t2times = []
+
+        for tim in timfiles:
+            tim = tempdir + "/" + tim  # add temporary directory path to file
+            print("Running PINT fitting w/o pickling...")
+            # run PINT w/o pickling and average time over 5 runs
+            pintrun(par_simple, tim, ptimes_nopickle, pickle=False, fitter="wls")
+
+            print("Running PINT w/ pickling...")
+            # run PINT with pickling and average time over 5 runs
+            subprocess.check_call(
+                "pintempo --usepickle " + par_simple + " " + tim,
+                shell=True,
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL,
+            )  # create pickle file
+            pintrun(par_simple, tim, ptimes_pickle, pickle=True, fitter="wls")
+            print("Running TEMPO...")
+            temporun(par_simple, tim, ttimes, fitter="wls")
+            print("Running Tempo2...")
+            tempo2run(par_simple, tim, t2times)
+
+        # create table 7 in PINT paper
+        simple_comparison = Table(
+            (ntoas_simple, ttimes, t2times, ptimes_nopickle, ptimes_pickle),
+            names=(
+                "Number of TOAs",
+                "TEMPO (sec)",
+                "Tempo2 (sec)",
+                "PINT - No Pickle (sec)",
+                "PINT - Pickle (sec)",
+            ),
+        )
+        ascii.write(
+            simple_comparison,
+            "simple_tables.pdf",
+            Writer=ascii.Latex,
+            latexdict={"tabletype": "table*"},
+            overwrite=True,
+        )
+
+        # time the individual PINT functions
+        importtimes = []
+        getTOAs_nopickle = []
+        getTOAs_pickle = []
+        fittimes = []
+
+        # time import statements
+        total = 0
+        for i in range(MAXIT):
+            start = timeit.default_timer()
+            subprocess.check_call(
+                "python3 import_statements.py",
+                shell=True,
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL,
+            )
+            end = timeit.default_timer()
+            total = total + (end - start)
+        for i in range(len(ntoas_simple)):
+            importtimes.append(total / MAXIT)
+
+        # setup
+        m = models.get_model(par_simple)
+        use_planets = False
+        if m.PLANET_SHAPIRO.value:
+            use_planets = True
+        model_ephem = "DE421"
+        if m.EPHEM is not None:
+            model_ephem = m.EPHEM.value
+
+        for tim in timfiles:
+            tim = tempdir + "/" + tim  # add temporary directory path to file
+            # no pickle time of get_TOAs
+            print("timing get_TOAs w/o pickling...")
+            total = 0
+            for i in range(MAXIT):
+                start = timeit.default_timer()
+                toa.get_TOAs(
```

Comment:
You can use the (new) `model` argument to `get_TOAs` to avoid having to pull the ephemeris and planets setting and so forth out of the model yourself.